gemini.lisp

```diff
@@ -49,14 +49,22 @@
       (error 'malformed-response :reason "missing meta"))
     (list (parse-status status) meta)))
 
-(defun read-all-stream (in)
+(defun read-all-string (in)
   (with-output-to-string (out)
     (loop with buffer = (make-array 1024 :element-type 'character)
           for n-chars = (read-sequence buffer in)
           while (< 0 n-chars)
           do (write-sequence buffer out :start 0
                                         :end n-chars))))
 
+(defun read-all-bytes (in)
+  (let ((data (make-array 0 :element-type '(unsigned-byte 8) :adjustable t :fill-pointer 0)))
+    (loop with buffer = (make-array 1024 :element-type '(unsigned-byte 8))
+          for n-bytes = (read-sequence buffer in)
+          while (< 0 n-bytes)
+          do (serapeum:vector-conc-extend data buffer n-bytes))
```

Comment:
I've yet to test the diff (will do soonish), is `serapeum` already a dependency?

Reply:
Oh wait, it's not! Let me look at what `serapeum` does under the hood and mimick it then :)